### DISCUSSION TITLE
When poweron in shutter mode tasmota set LOW level on Relay_i (GRIO5), but must set HIGH

### DISCUSSION THREAD

**Discussion by tantra35** [arendst/Tasmota#21997] [Q&A] [answered]
### PROBLEM DESCRIPTION
We use ESP-12F flashed with tasmota + a4988 stepper motor driver, and found follow issue: when we switch on on `ENABLE` pin(on esp12f this is GPIO5 in out case) of a4988 we see `LOW` level, due this a4988(with `LOW` level on `ENABLE` pin a4988 is activated) pass voltage on coils of motor, that follow to overheat motor 

after any actions of motor is making(for example shutter close/open), level of ENABLE pin set to HIGH, which disables a4988, and coils of motor not under voltage. This situation repeat again and again when switch power Off/On

To workaround the problem we set PowerOnState to 1, but its not very safety, because when power appear this can lead to unpredictable movements of shutters, moreover, these movements are rather undesirable(because this reset desire shutter position. in our case when switch off/on, shutters always closing)

### TO REPRODUCE
Just switch on/off, and check ENABLE pin (Relay_i - inverted relay - in our case in on GPIO5)

### EXPECTED BEHAVIOUR
When switch on on Relay_i pin must be set `HiGH` level not `LOW`

**Comment by sfromis** [comment]
Well, when powering up, there will by necessity be a brief moment until the ESP8266 boots Tasmota, before there is a chance of doing anything to control pin states. If you cannot live with the pin not being high (floating, not the same as being low) until `PowerOnState` kicks in, the solution has to be in hardware, not in Tasmota. Maybe it could work with a "sufficiently weak" pull-up resistor.

Ideally, you should not be cutting power to the ESP8266, as you then lose the chance to exert any control.

**Comment by tantra35** [comment]
@sfromis you are right pull up resistor to +3.3V power line helps solve problem, but this absolutely not obvious, i thinked that GPIO may be in only in 2 states `LOW/HIGH`, but situation that I discover show that really 3 states may present on GPIO pins after full tasmota initialisation - `LOW/HIGH/NOT CONNECTED`

I made some experiments, and found that sometimes HIGH level on `Relay_i` pin may present after switch on, but this happens absolutely randomly, about 1-2 times in 10 times 

and this strange for me `NOT CONNECTED` state may present on pin for a long time(not brief moment as you write). In our case happens follow there was damage on lines of electric company, so full house was without electricity, then after 1 hours, power appear again, and all shutters turned on, and of some of them happens situation that i describe, that state was present  in multiple hours(because there were no actions that would be required by automation), that led some plastic holder of motors are damaged(they melted a little)

At the beginning, I thought since that pin goes to HIGH state after some action, that the problem is on software side, and this strange third state `NOT CONNECTED` looks for me like something incredible(I'm not expert in PCB)

**Reply by sfromis** [reply, marked as answer]
Floating is a normal and very common state until there is an active control to drive a pin high or low. It is possible that a circuit may not be designed to handle floating, meaning that it might not react in a predictable and consistent way. I can't say if this applies in your build. With normal non-inverted pin usage, my starting point would be to expect it to be handled as low, but I have no clear expectations for the inverted case.

And when I mentioned "brief", that was until you can have Tasmota enact what you want, like depending on what `PowerOnState` you have.

Anyway, taking control over "unenforced" state (to avoid floating) with a pull-up or pull-down resistor is a standard part of circuit design.

Added: An important reason for having the "floating" state of a physically connected pin is that the MCU cannot know if the pin is for output or input until the firmware has booted, and able to act on its configuration. Unconfigured pin will of course the left floating after boot.

**Comment by stefanbode** [comment]
This is the reason you should have a circuit safe configuration as it is described here: https://tasmota.github.io/docs/Blinds-and-Shutters/#using-normal-motors

With this relay configuration you physically ensure the motor will never get power on both lines (up/down). If you buy a device like a shelly they take care the relay start OFF on boot until the software takes over. Also this is save and works fine. Before connecting the motors I always do a "restart test" and check if I can hear a minimum relay click. If yes, this IS A MAJOR problem. In most cases it helps to use the RELAY_I and change the cabeling accordingly.
